### ISSUE TITLE
Thumbnails: Apple Display P3 profile support for more accurate colors

### ISSUE THREAD

**Issue by lastzero** [photoprism/photoprism#1798]
*sRGB* used to be the only relevant standard for the Web (computer screens) before Apple started to push their own *Display P3* profile. The print and the movie industry traditionally have other standards suitable for printers and professional use cases.

**To preserve original colors as good as possible on all *sRGB* capable devices, [this change](https://github.com/photoprism/photoprism/commit/5be456a09f7a846d233649c8de21dbb7f4e1a45a) converts Apple *Display P3* to standard *sRGB* in JPEG thumbnails.**

As *sRGB* still is the default for all Web browsers, it doesn't require additional file headers or other metadata to be displayed as intended.

- A slight difference may be visible on high-end (Apple) devices
- Tests on a normal mid-range Dell screen show no noticeable difference in perception - unlike before
- For most users, this should be a major improvement

In a future release, PhotoPrism may render thumbnails with custom ICC Color Profiles and embed them. To make this possible, the follow-up issue #1474 lists a number of research tasks that should be completed beforehand.

**Comment by lastzero** [comment]
See also:
- #1522

**Comment by lastzero** [comment]
Related Publications and Libraries:
- https://github.com/mandykoh/prism
- https://blog.johnnovak.net/2016/09/21/what-every-coder-should-know-about-gamma/
- https://www.benq.com/en-us/knowledge-center/knowledge/gamma-monitor.html
- https://entropymine.com/imageworsener/gamma/
- http://www.ericbrasseur.org/gamma.html?i=1
- https://www.ingentaconnect.com/contentone/ist/cic/2020/00002020/00000028/art00031
- https://www.hdm-stuttgart.de/~froehlichj/diss/Jan_Froehlich_Thesis_2018_02_15_Acrobat_300dpi.pdf

**Comment by lastzero** [comment]
For reference, these are "actual" color profiles PhotoPrism found in my personal photo library:

```
+---------------------------------+
| file_color_profile              |
+---------------------------------+
| Adobe RGB (1998)                |
| AdobeRGB                        |
| Apple RGB                       |
| Camera RGB Profile              |
| Canon EOS 5D                    |
| Color LCD Calibrated            |
| DELL U2515H                     |
| Display                         |
| Display P3                      |
| Dot Gain 15%                    |
| Dot Gain 20%                    |
| EPSON  Standard RGB - Gamma 1.8 |
| GIMP built-in sRGB              |
| Generic RGB Profile             |
| Google Skia                     |
| Gray Gamma 2.2                  |
| HD 709-A                        |
| PHL 271E1                       |
| ProPhoto RGB                    |
| S2431W                          |
| c2                              |
| iMac                            |
| sRGB                            |
| sRGB IEC61966-2-1 black scaled  |
| sRGB IEC61966-2.1               |
| sRGB Transfer with DCI-P3 Gamut |
| sRGB built-in                   |
+---------------------------------+
```
